src/layers/core/scatterplot-layer/scatterplot-layer-vertex.glsl.js

```diff
@@ -60,9 +60,10 @@ void main(void) {
   vec3 vertex = positions * outerRadiusPixels;
   gl_Position = project_to_clipspace(vec4(center + vertex, 1.0));
 
+  // Set color to be rendered to picking fbo which is used to check for selection highlight.
+  picking_setPickingColor(instancePickingColors);
+
   // Apply opacity to instance color, or return instance picking color
-  vec4 color = vec4(instanceColors.rgb, instanceColors.a * opacity) / 255.;
-  vec4 pickingColor = vec4(instancePickingColors / 255., 1.);
-  vColor = mix(color, pickingColor, renderPickingBuffer);
+  vColor = vec4(instanceColors.rgb, instanceColors.a * opacity) / 255.;
```

Comment:
Do we have any perf measurements showing the impact of moving the check to the fragment shader?